tests/conftest.py

```diff
@@ -19,137 +22,321 @@ def app(request):
+    _framework_fixture_inner(request, app, **dict(
+        _g8_framework_defaults,
+        status="live",
+        framework_agreement_details=dict(_generic_framework_agreement_details, variations={
+            "banana": {
+                "createdAt": "2016-06-06T20:01:34.000000Z",
+            },
+            "toblerone": {
+                "createdAt": "2016-07-06T21:09:09.000000Z",
+            },
+        }),
+    ))
 
 
 # G6
 
 
 @pytest.fixture()
 def open_g6_framework(request, app):
-    _g6_framework(request, app, status='open')
+    _framework_fixture_inner(request, app, **dict(_g6_framework_defaults, status="open"))
 
 
 @pytest.fixture()
 def expired_g6_framework(request, app):
-    _g6_framework(request, app, status='expired')
+    _framework_fixture_inner(request, app, **dict(_g6_framework_defaults, status="expired"))
 
 
 # DOS
 
 
 @pytest.fixture()
 def open_dos_framework(request, app):
-    _dos_framework(request, app, status='open')
+    _framework_fixture_inner(request, app, **dict(_dos_framework_defaults, status="open"))
 
 
 @pytest.fixture()
 def live_dos_framework(request, app):
-    _dos_framework(request, app, status='live')
+    _framework_fixture_inner(request, app, **dict(_dos_framework_defaults, status="live"))
 
 
 @pytest.fixture()
 def expired_dos_framework(request, app):
-    _dos_framework(request, app, status='expired')
+    _framework_fixture_inner(request, app, **dict(_dos_framework_defaults, status="expired"))
+
+
+# Suppliers
+
+
+@pytest.fixture()
+def supplier_basic(request, app):
+    return _supplier_fixture_inner(request, app, supplier_kwargs={"supplier_id": 1})
+
+
+@pytest.fixture()
+def supplier_basic_alt(request, app):
+    return _supplier_fixture_inner(request, app, supplier_kwargs={"supplier_id": 2})
+
+
+# Users
+
+
+@pytest.fixture()
+def user_role_supplier(request, app, supplier_basic):
+    return _user_fixture_inner(request, app, user_kwargs={
+        "id": 1,
+        "role": "supplier",
+        "supplier_id": supplier_basic,
+    })
+
+
+@pytest.fixture()
+def user_role_supplier_alt(request, app, supplier_basic_alt):
+    return _user_fixture_inner(request, app, user_kwargs={
+        "id": 2,
+        "role": "supplier",
+        "supplier_id": supplier_basic_alt,
+    })
+
+
+# Frameworks & Suppliers with some SupplierFrameworks setup
+
+
+@pytest.fixture()
+def live_g8_framework_2_variations_suppliers_not_on_framework(
+        request,
+        app,
+        live_g8_framework_2_variations,
+        user_role_supplier,
+        ):
+    _supplierframework_fixture_inner(request, app)
+
+
+@pytest.fixture()
+def live_g8_framework_2_variations_suppliers_on_framework(
+        request,
+        app,
+        live_g8_framework_2_variations,
+        user_role_supplier,
+        ):
+    _supplierframework_fixture_inner(request, app, sf_kwargs={"on_framework": True})
+
+
+@pytest.fixture()
+def live_g8_framework_2_variations_suppliers_on_framework_with_alt(
```

Comment:
Yes... you can see I was already finding the fixture names ridiculously long and trying to find ways to compress them...